### PR TITLE
sh4d0wup 0.9.0 (new formula)

### PR DESCRIPTION
**Pull request by kpcyrd**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/sh4d0wup.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  depends_on "llvm@15" => :build # for libclang
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on "nettle"
+  depends_on "openssl@3"
+  depends_on "pcsc-lite"
+  depends_on "zstd"
+
+  def install
+    system "cargo", "install", *std_cargo_args
```
Comment (chenrui333) [inline Formula/sh4d0wup.rb:17]:
```suggestion
    # Ensure that the `openssl` crate picks up the intended library.
    ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix
    ENV["OPENSSL_NO_VENDOR"] = "1"

    system "cargo", "install", *std_cargo_args
```

File: Formula/sh4d0wup.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    generate_completions_from_executable(bin/"sh4d0wup", "completions")
+  end
+
+  test do
+    system bin/"sh4d0wup", "keygen", "tls"
+    system bin/"sh4d0wup", "keygen", "pgp"
+    system bin/"sh4d0wup", "keygen", "ssh", "--type=ed25519", "--bits=256"
+    system bin/"sh4d0wup", "keygen", "openssl", "--secp256k1"
+  end
```
Comment (chenrui333) [inline Formula/sh4d0wup.rb:27]: can we also assert the response in here?

Also we probably need linkage test as well

```
  test do
    linked_libraries = [
      Formula["openssl@3"].opt_lib/shared_library("libssl"),
      Formula["openssl@3"].opt_lib/shared_library("libcrypto"),
    ]
    linked_libraries << (Formula["openssl@3"].opt_lib/shared_library("libcrypto")) if OS.mac?

    linked_libraries.each do |library|
      assert check_binary_linkage(bin/"sh4d0wup", library),
             "No linkage with #{library.basename}! Cargo is likely using a vendored version."
    end
  end
```

File: Formula/sh4d0wup.rb (hunk 1 above)
Comment (kpcyrd) [reply Formula/sh4d0wup.rb:21]: Thanks, I've applied this!

File: Formula/sh4d0wup.rb (hunk 2 above)
Comment (kpcyrd) [reply Formula/sh4d0wup.rb:55]: > Is there a way to pipe the output of my other commands into a program (like pgpdump or openssl x509)? I'm not sure how to do this in ruby. :)

I think I got it to work! :3

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@kpcyrd, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**